packages/plugin-conversation/src/conversation.js

```diff
@@ -0,0 +1,607 @@
+/**!
+ *
+ * Copyright (c) 2015-2016 Cisco Systems, Inc. See LICENSE file.
+ * @private
+ */
+
+import {SparkPlugin} from '@ciscospark/spark-core';
+import {defaults, isArray, isObject, isString, last, map, merge, omit, pick, uniq} from 'lodash';
+import Decrypter from './decrypter';
+import Encrypter from './encrypter';
+import Normalizer from './normalizer';
+import uuid from 'uuid';
+import querystring from 'querystring';
+
+const Conversation = SparkPlugin.extend({
+  namespace: `Conversation`,
+
+  children: {
+    decrypter: Decrypter,
+    encrypter: Encrypter,
+    normalizer: Normalizer
+  },
+
+  add(conversation, object, activity) {
+    return this._inferConversationUrl(conversation)
+      .then(() => this.spark.user.asUUID(object, {create: true}))
+      .then((id) => this.prepare(activity, {
+        verb: `add`,
+        target: this.prepareConversation(conversation),
+        object: {
+          id,
+          objectType: `person`
+        },
+        kmsMessage: {
+          method: `create`,
+          uri: `/authorizations`,
+          resourceUri: `<KRO>`,
+          userIds: [
+            id
+          ]
+        }
+      })
+      .then((a) => this.submit(a)));
+  },
+
+  create(params, options) {
+    if (!params.participants || params.participants.length === 0) {
+      return Promise.reject(new Error(`\`params.participants\` is required`));
+    }
+
+    return Promise.all(params.participants.map((participant) => this.spark.user.asUUID(participant, {create: true})))
+      .then((participants) => {
+        participants.unshift(this.spark.device.userId);
+        params.participants = uniq(participants);
+
+        if (participants.length === 2 && !(options && options.forceGrouped)) {
+          return this._maybeCreateOneOnOneThenPost(params);
+        }
+
+        return this._createGrouped(params);
+      });
+  },
+
+  expand(verb, object, target, actor) {
+    const activity = {
+      actor,
+      objectType: `activity`,
+      verb
+    };
+
+    if (!actor) {
+      actor = this.spark.device.userId;
+    }
+
+    if (isString(actor)) {
+      activity.actor = {
+        objectType: `person`,
+        id: actor
+      };
+    }
+
+    if (object) {
+      activity.object = object;
+    }
+
+    if (target) {
+      activity.target = target;
+    }
+
+    return activity;
+  },
+
+  get(conversation, options) {
```

Comment:
If there's any function that could probably use some docs it's this one :) maybe you did it in another PR though.